MSAL Angular SPA (#15)
Azure-Samples/ms-identity-docs-code-javascript: angular-spa/README.md

| **Redirect URI**                  | `http://localhost:4200`                                                      | Required value for this sample.                                                                    |

> :information_source: **Bold text** in the tables above matches (or is similar to) a UI element in the Azure portal, while `code formatting` indicates a value you enter into a text box in the Azure portal.

### 2. Update code sample in _app.module.ts_ with app registration values

```javascript
// 'Application (client) ID' of app registration in Azure portal - this value is a GUID
clientId: '',

// Full directory URL, in the form of https://login.microsoftonline.com/<tenant>
authority: '',

// Must be the same redirectUri as what was provided in your AD app registration.
redirectUri: ''
```

### 3. Install package(s)

To install required packages:

```console
npm install
```

## Run the application

```console
ng serve --open
```

Follow the instructions that are presented on the web browser. If everything worked, you should receive a response similar to this on the "View Profile" page:

![Screenshot with profile information from Graph](profile.png?raw=true)

> ## About the code

Reviewer: Please note the filenames each of these are used in.

> This Angular application uses the Angular MSAL v2 library to authenticate a user and then makes a request to the Graph API as the authenticated user.

Reviewer: ```suggestion
This Angular web application uses the Angular MSAL v2 library to authenticate a user and then makes a request to the Microsoft Graph API as the authenticated user to retrieve their user profile data.
```

> - MSALGuard is used to protect routes

Reviewer: ```suggestion
- MSALGuard is used to protect the web app's routes.
```

Also, which routes?

> - MSALBroadcastService is used to consume events emitted by MSAL

Reviewer: Which event(s) and why?

> - MSALInterceptor is provided by MSAL Angular to automatically acquire tokens for protected resources

Reviewer: Access tokens, ID tokens, or both?

> - HttpClient is used to perform an HTTP GET request against Graph with the user's token

Reviewer: ```suggestion
- HttpClient is used to issue an HTTP GET request to Microsoft Graph with user's access token.
```
